src/runner/restart_elevated.cpp

```diff
@@ -0,0 +1,35 @@
+#include "pch.h"
+#include "restart_elevated.h"
+#include "common/common.h"
+
+enum State {
+  None,
+  RestartAsElevated,
+  RestartAsNonElevated
+};
+static State state = None;
+
+void schedule_restart_as_elevated() {
+  state = RestartAsElevated;
+}
+
+void schedule_restart_as_non_elevated() {
+  state = RestartAsNonElevated;
+}
+
+bool is_restart_scheduled() {
+  return state != None;
+}
+
+bool restart_if_scheduled() {
+  std::array<wchar_t, 0xFFFF> exe_path;

```

Comment:
why 0xFFFF instead of MAX_PATH?

Reply:
https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file#maximum-path-length-limitation

Unfortunately, calling GetModuleFileNameW with zero size will not yield the required buffer length like most other APIs. 0xFFFF will definitely be enough for the file name.